### DISCUSSION TITLE
client_protocol is unset in DBeaver 3.6

### DISCUSSION THREAD

**Discussion by ghost** [dbeaver/dbeaver#212] [Q&A] [answered]
Starting with DBeaver 3.6 I get the following error message on a connection that worked with older versions:

> SQLException: Could not establish connection to jdbc:hive2://**_secret**_.lan:10000/default: Required field 'client_protocol' is unset! Struct:TOpenSessionReq(client_protocol:null, configuration:{use:database=default})
>   org.apache.thrift.TApplicationException: TApplicationException: Required field 'client_protocol' is unset! Struct:TOpenSessionReq(client_protocol:null, configuration:{use:database=default})

As a result I can not connect to hive.

Unfortunately there are no further details in the error log.

**Comment by serge-rider** [comment]
This is not a DBeaver issue but some misconfiguration in the new driver. Most likely your client driver is not compatible with server version.
Check this: http://stackoverflow.com/questions/24694415/required-field-client-protocol-is-unset
You could try to use previous Hive JDBC driver version (1.2.1 or earlier). If you use driver in Maven artifact then you can change it's version in download dialog.

**Comment by ghost** [comment]
That worked, thanks a lot!

**Comment by mverrilli** [comment]
I know this is old, but I ran into the same issue. The JIRA for it is: https://issues.apache.org/jira/browse/HIVE-6050

Basically they didn't make it backwards compatible, so new client won't work with old server. It's been an open issue since 2013 so it's safe to say it'll eventually become a "won't fix" :-)

I didn't realize I could change it in the Download dialog, thanks.

**Comment by serge-rider** [comment, marked as answer]
Just click on version number and change it:
![default](https://cloud.githubusercontent.com/assets/6398845/26742566/db9ed2e6-47e6-11e7-965b-1eead4903bba.png)
